src/main/java/com/blackducksoftware/integration/alert/common/AlertProperties.java

```diff
@@ -0,0 +1,178 @@
+ * under the License.
+ */
+package com.blackducksoftware.integration.alert.common;
+
+import java.util.Optional;
+import java.util.Properties;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.blackducksoftware.integration.hub.configuration.HubServerConfigBuilder;
+import com.blackducksoftware.integration.log.IntLogger;
+import com.blackducksoftware.integration.rest.connection.UnauthenticatedRestConnectionBuilder;
+import com.blackducksoftware.integration.rest.proxy.ProxyInfoBuilder;
+
+@Component
+public class AlertProperties {
+    @Value("${alert.config.home:}")
+    private String alertConfigHome;
+
+    @Value("${alert.templates.dir:}")
+    private String alertTemplatesDir;
+
+    @Value("${alert.images.dir:}")
+    private String alertImagesDir;
+
+    @Value("${alert.trust.cert:}")
+    private Boolean alertTrustCertificate;
+
+    @Value("${alert.proxy.host:}")
+    private String alertProxyHost;
+
+    @Value("${alert.proxy.port:}")
+    private String alertProxyPort;
+
+    @Value("${alert.proxy.username:}")
+    private String alertProxyUsername;
+
+    @Value("${alert.proxy.password:}")
+    private String alertProxyPassword;
+
+    public Optional<String> getAlertConfigHome() {
+        if (StringUtils.isNotBlank(alertConfigHome)) {
+            return Optional.of(alertConfigHome);
+        }
+        return Optional.empty();
+    }
+
+    public Optional<String> getAlertTemplatesDir() {
+        if (StringUtils.isNotBlank(alertTemplatesDir)) {
+            return Optional.of(alertTemplatesDir);
+        }
+        return Optional.empty();
+    }
+
+    public Optional<String> getAlertImagesDir() {
+        if (StringUtils.isNotBlank(alertImagesDir)) {
+            return Optional.of(alertImagesDir);
+        }
+        return Optional.empty();
+    }
+
+    public Optional<Boolean> getAlertTrustCertificate() {
+        return Optional.ofNullable(alertTrustCertificate);
+    }
+
+    public Optional<String> getAlertProxyHost() {
+        if (StringUtils.isNotBlank(alertProxyHost)) {
+            return Optional.of(alertProxyHost);
+        }
+        return Optional.empty();
+    }
+
+    public Optional<String> getAlertProxyPort() {
+        if (StringUtils.isNotBlank(alertProxyPort)) {
+            return Optional.of(alertProxyPort);
+        }
+        return Optional.empty();
+    }
+
+    public Optional<String> getAlertProxyUsername() {
+        if (StringUtils.isNotBlank(alertProxyUsername)) {
+            return Optional.of(alertProxyUsername);
+        }
+        return Optional.empty();
+    }
+
+    public Optional<String> getAlertProxyPassword() {
+        if (StringUtils.isNotBlank(alertProxyPassword)) {
+            return Optional.of(alertProxyPassword);
+        }
+        return Optional.empty();
+    }
```

Comment:
For all these string optionals, it might be convenient to have a getStringOptional private method which takes in the string field and does this repeated code for us then just call this method for each getter.